yarn-project/noir-protocol-circuits/src/index.test.ts

```diff
@@ -68,228 +25,32 @@ import { dirname, resolve } from 'path';
-      Fr.ZERO,
-      Fr.ZERO,
-    );
-
-    const callContext = new CallContext(AztecAddress.ZERO, contractAddress, Fr.ZERO, selector, false, false, true, 0);
-
-    const blockHeader = new BlockHeader(
-      Fr.fromString('0x16642d9ccd8346c403aa4c3fa451178b22534a27035cdaa6ec34ae53b29c50cb'),
-      Fr.fromString('0x0bcfa3e9f1a8922ee92c6dc964d6595907c1804a86753774322b468f69d4f278'),
-      Fr.fromString('0x1864fcdaa80ff2719154fa7c8a9050662972707168d69eac9db6fd3110829f80'),
-      Fr.fromString('0x1864fcdaa80ff2719154fa7c8a9050662972707168d69eac9db6fd3110829f80'),
-      Fr.fromString('0x1759d221795419503f86c032e8f8762f2b739e74835099584b6531f5f27390fe'),
-      Fr.ZERO, // TODO(#3441)
-      Fr.fromString('0x0ccaafdc9c353743970d4e305ae73641ce694f07db67886d2769c9ed88e969d8'),
-      Fr.fromString('0x200569267c0f73ac89aaa414239398db9445dd4ad3a8cf37015cd55b8d4c5e8d'),
-    );
-
-    const appPublicInputs = new PrivateCircuitPublicInputs(
-      callContext,
-      argsHash,
-      makeTuple(RETURN_VALUES_LENGTH, () => Fr.ZERO),
-      makeTuple(MAX_READ_REQUESTS_PER_CALL, () => SideEffect.empty()),
-      makeTuple(MAX_NULLIFIER_KEY_VALIDATION_REQUESTS_PER_CALL, () => NullifierKeyValidationRequest.empty()),
-      newCommitments,
-      newNullifiers,
-      makeTuple(MAX_PRIVATE_CALL_STACK_LENGTH_PER_CALL, () => Fr.ZERO),
-      makeTuple(MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL, () => Fr.ZERO),
-      makeTuple(MAX_NEW_L2_TO_L1_MSGS_PER_CALL, () => Fr.ZERO),
-      Fr.ZERO,
-      [Fr.fromString('0x9cc0744c0dde14f24854659b052ffb7e'), Fr.fromString('0x28120e19a5cc9ec344f3d6d41b6fada2')],
-      [Fr.fromString('0xe3b0c44298fc1c149afbf4c8996fb924'), Fr.fromString('0x27ae41e4649b934ca495991b7852b855')],
-      Fr.fromString('0xf8'),
-      Fr.fromString('0x04'),
-      blockHeader,
-      contractDeploymentData,
-      Fr.ZERO,
-      Fr.ZERO,
-    );
-
-    const callStackItem = new PrivateCallStackItem(contractAddress, functionData, appPublicInputs, false);
+    const filepath = resolve(dirname(fileURLToPath(import.meta.url)), './fixtures/nested-call-private-kernel-init.hex');
+    const serialized = Buffer.from(readFileSync(filepath).toString(), 'hex');
+    const kernelInputs = PrivateKernelInputsInit.fromBuffer(serialized);
 
-    const privateCall = new PrivateCallData(
-      callStackItem,
-      makeTuple(MAX_PRIVATE_CALL_STACK_LENGTH_PER_CALL, () => CallRequest.empty()),
-      makeTuple(MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL, () => CallRequest.empty()),
-      makeEmptyProof(),
-      VerificationKey.makeFake(),
-      MembershipWitness.empty(FUNCTION_TREE_HEIGHT, 0n),
-      MembershipWitness.empty(CONTRACT_TREE_HEIGHT, 0n),
-      makeTuple(MAX_READ_REQUESTS_PER_CALL, () => ReadRequestMembershipWitness.empty(0n)),
-      Fr.ZERO,
-      Fr.ZERO,
-    );
-    const kernelInputs = new PrivateKernelInputsInit(txRequest, privateCall);
+    if (!kernelInputs.txRequest.txContext.isContractDeploymentTx) {
```

Comment:
Why is this throwing an error instead of checking with an `expect()`?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/4235/commits/bba328d97cb5d2e6266a1ef702a68bc24b981a33